app/policies/user_policy.rb

```diff
@@ -12,7 +12,12 @@ def create?
   end
 
   def update?
-    permissions?
+    # A user with role 'admin' is not allowed to edit their own role
+    if user == record
+      user.role != User::ADMIN
```

Comment:
you can also use `user.admin?` to avoid the coupling how the role is implemented (i.e. with a string 'admin')

Reply:
or, in this case, `!user.admin?`